src/Prefix.jl

```diff
@@ -477,7 +477,9 @@ function get_addable_spec(name::AbstractString, version::VersionNumber;
 
         LibGit2.with(LibGit2.GitRepo(dir)) do repo
             LibGit2.with(LibGit2.GitRevWalker(repo)) do walker
-                LibGit2.push_head!(walker)
+                # The repo is cached, so locally it may be checking out an outdated commit.
+                # Start the search from HEAD of the tracking upstream repo.
+                LibGit2.push!(walker, LibGit2.GitHash(LibGit2.peel(LibGit2.GitCommit, LibGit2.upstream(LibGit2.head(repo)))))
```

Comment:
```suggestion
                try
                    LibGit2.push!(walker, LibGit2.GitHash(LibGit2.peel(LibGit2.GitCommit, LibGit2.upstream(LibGit2.head(repo)))))
                catch
                    @warn("Could not walk from origin branch!")
                    LibGit2.push_head!(walker)
                end
```

Let's look out for cases where something goes wrong.